### PR TITLE
[BACK-2679] deps updates 

### PR DESCRIPTION
**Pull request by jh-bate**
Done:
- remove unused deps
- bump node version
- bump mongo version 
- `npm outdated`
- `npm audit fix`

TODO:
- updates from internal deps

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,11 +3,11 @@ dist: focal
 language: node_js
 
 node_js:
-  - 14.21.2
+  - 16.20.1
```
Comment (darinkrauss) [inline .travis.yml:6]: We should probably decide which version we will use for all of the backend Node.js repos. For the repos I updated I migrated directly to `18.18.1` as any earlier versions are EOL (according to https://nodejs.dev/en/about/releases/).

File: .travis.yml [hunk 2]
```diff
@@ -3,11 +3,11 @@ dist: focal
...
 
 env:
   global:
-  - MONGODB=5.3.2
+  - MONGODB=6.0.10
```
Comment (darinkrauss) [inline .travis.yml:10]: Does this match our current production version? If so, I'll update the repos I updated to this version, too.

File: Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ### Stage 0 - Base image
-FROM node:14.21.2-alpine as base
+FROM node:16.20.1-alpine as base
```
Comment (darinkrauss) [inline Dockerfile:2]: Same comment as above about Node.js version.

File: package.json [hunk 4]
```diff
@@ -16,31 +16,27 @@
...
-    "cookies": "0.8.0",
-    "lodash": "4.17.20",
-    "moment": "^2.29.0",
-    "mongodb": "^5.0.1",
-    "mongodb-legacy": "^5.0.0",
-    "request": "2.88.2",
-    "restify": "8.5.1",
-    "sundial": "1.6.0",
+    "@godaddy/terminus": "4.12.1",
+    "amoeba": "1.0.2",
```
Comment (darinkrauss) [inline package.json:20]: Will require update to latest `amoeba` once published.

File: package.json [hunk 5]
```diff
@@ -16,31 +16,27 @@
...
+    "bunyan": "1.8.15",
+    "lodash": "4.17.21",
+    "moment": "2.29.4",
+    "mongodb-legacy": "6.0.0",
+    "sundial": "1.7.1",
     "tidepool-gatekeeper": "0.2.4",
```
Comment (darinkrauss) [inline package.json:26]: Will require updates to latest `tidepool-gatekeeper`, `tidepool-seagull-client`, and `user-api-client` once published.

File: lib/handler/mongoHandler.js [hunk 6]
```diff
@@ -152,7 +152,7 @@ module.exports = function (connectionString) {
       if (err) {
         return callback(err);
       }
-      const update = result.value;
+      const update = result;
```
Comment (darinkrauss) [inline lib/handler/mongoHandler.js:155]: This change seems concerning. Was this required by the mongo update or something else?

**Review by darinkrauss — CHANGES_REQUESTED** [review]
A question or two, but mostly just that other dependencies will require updating.

File: .travis.yml (hunk 2 above)
Comment (jh-bate) [reply .travis.yml:10]: I updated this based on from your question  in the backed channel. Jimmy said the current Mongo version was `6.0.10`

File: lib/handler/mongoHandler.js (hunk 6 above)
Comment (jh-bate) [reply lib/handler/mongoHandler.js:155]: yes - it was from the mongo update

File: .travis.yml (hunk 1 above)
Comment (jh-bate) [reply .travis.yml:6]: sounds good - looks like it is now `18.18.2`

**Pull request closed without merging**